src/main/java/bral/dictionary/DictionaryFrame.java

```diff
@@ -0,0 +1,89 @@
+package bral.dictionary;
+
+import com.opencsv.exceptions.CsvValidationException;
+import javax.swing.*;
+import java.awt.*;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import java.io.IOException;
+import java.util.List;
+
+public class DictionaryFrame extends JFrame
+{
+
+    private final EnglishDictionary englishDictionary = new EnglishDictionary();
+
+    private final DefinitionsComponent definitionsComponent;
+
+    private JTextField searchField;
+
+    private JTextArea definitionsTextArea;
+
+    public DictionaryFrame() throws CsvValidationException, IOException
+    {
+        setSize(1000, 800);
+        setTitle("English Dictionary");
+
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        JPanel main = new JPanel();
+        main.setLayout(new BorderLayout());
+        // tells the JFrame to use this JPanel
+        setContentPane(main);
+
+        definitionsComponent = new DefinitionsComponent(englishDictionary.getDefinition(""));
+        add(definitionsComponent, BorderLayout.CENTER);
+
+        searchField = new JTextField("");
+        main.add(searchField, BorderLayout.NORTH);
+
+        definitionsTextArea = new JTextArea("Definitions will show here...");
+        definitionsTextArea.setEditable(false);
+        definitionsTextArea.setLineWrap(true);
+        definitionsTextArea.setWrapStyleWord(true);
+        main.add(definitionsTextArea, BorderLayout.CENTER);
+
+        JScrollPane scrollPane = new JScrollPane(definitionsTextArea);
+        main.add(scrollPane, BorderLayout.CENTER);
+
+        searchField.getDocument().addDocumentListener((new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                reDisplay();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                reDisplay();
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                reDisplay();
+            }
+
+        }));
+
+    }
+
+    private void reDisplay()
+    {
+        try {
+            String currWord = searchField.getText();
+            // update the definitionsComponent based on the current word:
+            List<String> currDefinitions = englishDictionary.getDefinition(currWord);
+            // definitionsComponent.setCurrDefinitions(currDefinitions);
+
+            // clear the old contents:
+            definitionsTextArea.setText("");
+            for (String definition : currDefinitions)
+            {
+                definitionsTextArea.append(definition + "\n\n");
+            }
+        } catch (NullPointerException e) {
```

Comment:
Why?

Reply:
It won't work without try/catch since if the user types in a word that's not in the dictionary, it would freeze without this block

Reply:
Ok... But fix that without the try catch